homeassistant/components/moon/sensor.py

```diff
@@ -60,30 +151,117 @@ def device_class(self):
         return "moon__phase"
 
     @property
-    def native_value(self):
-        """Return the state of the device."""
-        if self._state < 0.5 or self._state > 27.5:
-            return STATE_NEW_MOON
-        if self._state < 6.5:
-            return STATE_WAXING_CRESCENT
-        if self._state < 7.5:
-            return STATE_FIRST_QUARTER
-        if self._state < 13.5:
-            return STATE_WAXING_GIBBOUS
-        if self._state < 14.5:
-            return STATE_FULL_MOON
-        if self._state < 20.5:
-            return STATE_WANING_GIBBOUS
-        if self._state < 21.5:
-            return STATE_LAST_QUARTER
-        return STATE_WANING_CRESCENT
+    def state(self):
+        """Return the phase of the Moon."""
+        return self.phase
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the Moon."""
+        transit = None
+        if (self.next_transit is not None) and (self.next_antitransit is not None):
+            transit = self.next_transit < self.next_antitransit
+        return {
+            STATE_ATTR_NEXT_RISING: self.next_rising,
+            STATE_ATTR_NEXT_SETTING: self.next_setting,
+            STATE_ATTR_NEXT_TRANSIT: self.next_transit,
+            STATE_ATTR_NEXT_ANTITRANSIT: self.next_antitransit,
+            STATE_ATTR_ALTITUDE: self.altitude,
+            STATE_ATTR_AZIMUTH: self.azimuth,
+            STATE_ATTR_RISING: transit,
+            STATE_ATTR_ABOVE_HORIZON: self.altitude >= 0,
+            STATE_ATTR_ILLUMINATION: self.illumination,
+            STATE_ATTR_DISTANCE: self.distance,
```

Comment:
In general, we want to reduce the number of attributes in Home Assistant. In general, these can be separate sensors instead; making it easier to use them in Lovelace dashboards and automations.